client/signup/steps/import-url/index.jsx

```diff
@@ -5,162 +5,211 @@
-		const { translate } = this.props;
-		const message = isValidUrl( value )
-			? ''
-			: translate( 'Please enter the full URL of your site.' );
+	getIsSiteImportableError = () => {
+		if ( ! this.props.isSiteImportableError ) {
+			return null;
+		}
+
+		const { isSiteImportableError, translate } = this.props;
 
-		this.props.setValidationMessage( message );
-	}
+		switch ( isSiteImportableError.code ) {
+			case 1000001:
+				return translate( 'This does not appear to be a valid URL or website address.' );
+			// @TODO differentiate an unreachable site from one that's not compatible.
+			case 1000002:
+				return translate(
+					"We're not able to reach that site at this time, or it's not compatible with our URL importer."
+				);
+		}
+
+		return translate( 'There was an error with the importer, please try again.' );
+	};
 
 	renderContent = () => {
-		const { isInputDisabled, urlInputValidationMessage, urlInputValue, translate } = this.props;
-		const { showValidation } = this.state;
+		const { isLoading, urlInputValue, translate } = this.props;
+		const { showUrlMessage } = this.state;
+		const urlMessage = this.getUrlMessage();
 
 		return (
-			<div className="import-url__wrapper">
-				<FormTextInputWithAction
-					placeholder={ translate( 'Please enter the full URL of your site.' ) }
-					action="Continue"
-					onAction={ this.handleAction }
-					label={ translate( 'URL' ) }
-					onChange={ this.handleInputChange }
-					defaultValue={ urlInputValue }
-					disabled={ isInputDisabled }
-					onBlur={ this.handleInputBlur }
-				/>
-				{ showValidation ? (
-					<FormInputValidation
-						text={ urlInputValidationMessage || translate( 'This URL is valid.' ) }
-						isError={ !! urlInputValidationMessage }
+			<Card className="import-url__wrapper site-importer__site-importer-pane" tagName="div">
+				<form
+					className="import-url__form site-importer__site-importer-url-input"
+					onSubmit={ this.handleSubmit }
+				>
+					<FormTextInput
+						placeholder="https://example.wixsite.com/mysite"
+						disabled={ isLoading }
+						defaultValue={ urlInputValue }
+						onChange={ this.handleInputChange }
+						onBlur={ this.handleInputBlur }
+						inputRef={ this.handleInputRef }
 					/>
-				) : (
-					<FormSettingExplanation>
-						{ translate( 'Please enter the full URL of your site.' ) }
-					</FormSettingExplanation>
-				) }
-			</div>
+					<FormButton disabled={ isLoading } busy={ isLoading } type="submit">
+						{ translate( 'Continue' ) }
+					</FormButton>
+				</form>
+				{ showUrlMessage &&
+					urlMessage && <FormInputValidation text={ urlMessage } isError={ !! urlMessage } /> }
+			</Card>
 		);
 	};
 
 	render() {
-		const { flowName, positionInFlow, signupProgress, stepName, translate } = this.props;
+		const { flowName, isLoading, positionInFlow, signupProgress, stepName, translate } = this.props;
+		const noticeText = translate( "Please wait, we're checking this URL." );
```

Comment:
Would it be overkill to mention that were checking whether or not the URL is importable? - that

It might actually be better to use the term 'site' too as that's a bit less technical.

Reply:
I like that idea, will modify.